profile setup fails due to invalid timezone

On the Midway cluster at the University of Chicago, it turns out that this code
https://github.com/aiidateam/aiida-core/blob/cc94a1e0127cd9cab0a4049e69a9fa31cb82941c/aiida/common/timezone.py#L24-L25

returns a timezone called `local` which when passed here
https://github.com/aiidateam/aiida-core/blob/6299cf6c961f27c64c0a146130f52adafcba9d96/aiida/backends/djsite/settings.py#L72

gives rise to an error
```
ValueError: Incorrect timezone setting: local
```
when the django migrations are running.

We should look into why django and tzlocal seem to have a different idea of which timezones are allowed, and whether we can provide a fallback (or a suggestion to fix it) so that profile setup does not simply fail.

Note: `export TIME_ZONE='America/Chicago'` as [suggested here](https://stackoverflow.com/a/27182580/1069467) did not help

Looking at the [implementation of tzlocal](https://github.com/regebro/tzlocal/blob/a117f96f975febd4d9d7b51c71bda65ab4424303/tzlocal/unix.py#L36), it turns out that `local` actually does *not* mean that the time zone is unknown.
It just means that the "zoneinfo name" is unknown.

As discussed with @sphuber , one solution is to actually delete this setting
https://github.com/aiidateam/aiida-core/blob/6299cf6c961f27c64c0a146130f52adafcba9d96/aiida/backends/djsite/settings.py#L72
since it only configures how django *displays* datetime objects; a functionality that we don't use in AiiDA.

Mocking the `local` timezone using 
```python
def get_current_timezone():
    import pytz
    with open('/etc/localtime', 'rb') as handle:
        tz = pytz.tzfile.build_tzinfo('local', handle)
    return tz
```
reveals that two more tests fail:

```
======================================================================
ERROR: test_timezone_addition_and_dir_correction (aiida.backends.tests.manage.backup.test_backup_script.TestBackupScriptUnit)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/leopold/Personal/Postdoc-MARVEL/repos/aiida/aiida_rmq/aiida/backends/tests/manage/backup/test_backup_script.py", line 249, in test_timezone_addition_and_dir_correction
    self._backup_setup_inst._read_backup_info_from_dict(backup_variables)
  File "/Users/leopold/Personal/Postdoc-MARVEL/repos/aiida/aiida_rmq/aiida/manage/backup/backup_base.py", line 140, in _read_backup_info_from_dict
    self._oldest_object_bk = ptimezone(curr_timezone).localize(self._oldest_object_bk)
  File "/Users/leopold/Applications/miniconda3/envs/aiida_rmq_py3/lib/python3.7/site-packages/pytz/__init__.py", line 181, in timezone
    raise UnknownTimeZoneError(zone)
pytz.exceptions.UnknownTimeZoneError: 'local'

======================================================================
FAIL: test_timezone_now (aiida.backends.tests.common.test_timezone.TimezoneTest)
Test timezone.now function.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/leopold/Personal/Postdoc-MARVEL/repos/aiida/aiida_rmq/aiida/backends/tests/common/test_timezone.py", line 31, in test_timezone_now
    self.assertGreaterEqual(from_tz, ref - delta)
AssertionError: datetime.datetime(2019, 11, 26, 7, 1, 37, 661575, tzinfo=<DstTzInfo 'local' CET+1:00:00 STD>) not greater than or equal to datetime.datetime(2019, 11, 26, 13, 0, 37, 661565, tzinfo=<UTC>)
```

I'll look into whether those can be fixed as well.

--- a/aiida/backends/djsite/settings.py
+++ b/aiida/backends/djsite/settings.py
@@ -68,7 +68,9 @@ MANAGERS = ADMINS
 # http://www.i18nguy.com/unicode/language-identifiers.html
 LANGUAGE_CODE = 'en-us'
 
-# Local time zone for this installation. Always choose the system timezone
+# Local time zone for this installation. Always choose the system timezone.
+# Note: This causes django to set the 'TZ' environment variable, which is read by tzlocal from then onwards.
+# See https://docs.djangoproject.com/en/2.2/ref/settings/#std:setting-TIME_ZONE
 TIME_ZONE = get_current_timezone().zone
 
 SITE_ID = 1

--- a/aiida/backends/tests/common/test_timezone.py
+++ b/aiida/backends/tests/common/test_timezone.py
@@ -23,9 +23,14 @@ class TimezoneTest(unittest.TestCase):
     """Tests for the timezone utility module."""
 
     def test_timezone_now(self):
-        """Test timezone.now function."""
+        """Test timezone.now function.
+
+        Check that the time returned by AiiDA's timezone.now() function is compatible
+        with attaching a timezone to a "naive" time stamp using timezone.make_aware().
+        """
         delta = datetime.timedelta(minutes=1)
         ref = timezone.now()
+
         from_tz = timezone.make_aware(datetime.datetime.fromtimestamp(time.time()))
         self.assertLessEqual(from_tz, ref + delta)
         self.assertGreaterEqual(from_tz, ref - delta)

--- a/aiida/common/timezone.py
+++ b/aiida/common/timezone.py
@@ -22,7 +22,14 @@ def get_current_timezone():
     :return: current timezone
     """
     from tzlocal import get_localzone
-    return get_localzone()
+    local = get_localzone()
+
+    if local.zone == 'local':
+        raise ValueError(
+            "Unable to detect name of local time zone. Please set 'TZ' environment variable, e.g."
+            " to 'Europe/Zurich'"
+        )
+    return local
 
 
 def now():
@@ -70,8 +77,10 @@ def make_aware(value, timezone=None, is_dst=None):
 
     if is_aware(value):
         raise ValueError('make_aware expects a naive datetime, got %s' % value)
+
     # This may be wrong around DST changes!
-    return value.replace(tzinfo=timezone)
+    # See http://pytz.sourceforge.net/#localized-times-and-date-arithmetic
+    return timezone.localize(value)
 
 
 def localtime(value, timezone=None):

--- a/aiida/manage/backup/backup_base.py
+++ b/aiida/manage/backup/backup_base.py
@@ -21,7 +21,6 @@ import logging
 from abc import abstractmethod, ABCMeta
 import six
 from dateutil.parser import parse
-from pytz import timezone as ptimezone
 
 from aiida.common import json
 from aiida.common import timezone as dtimezone
@@ -136,11 +135,11 @@ class AbstractBackup(object):
             try:
                 self._oldest_object_bk = parse(backup_variables.get(self.OLDEST_OBJECT_BK_KEY))
                 if self._oldest_object_bk.tzinfo is None:
-                    curr_timezone = str(dtimezone.get_current_timezone())
-                    self._oldest_object_bk = ptimezone(curr_timezone).localize(self._oldest_object_bk)
+                    curr_timezone = dtimezone.get_current_timezone()
+                    self._oldest_object_bk = dtimezone.get_current_timezone().localize(self._oldest_object_bk)
                     self._logger.info(
                         'No timezone defined in the oldest modification date timestamp. Setting current timezone (%s).',
-                        curr_timezone
+                        curr_timezone.zone
                     )
             # If it is not parsable...
             except ValueError:
@@ -168,13 +167,13 @@ class AbstractBackup(object):
                 self._end_date_of_backup = parse(backup_variables.get(self.END_DATE_OF_BACKUP_KEY))
 
                 if self._end_date_of_backup.tzinfo is None:
-                    curr_timezone = str(dtimezone.get_current_timezone())
+                    curr_timezone = dtimezone.get_current_timezone()
                     self._end_date_of_backup = \
-                        ptimezone(curr_timezone).localize(
+                        curr_timezone.localize(
                             self._end_date_of_backup)
                     self._logger.info(
                         'No timezone defined in the end date of backup timestamp. Setting current timezone (%s).',
-                        curr_timezone
+                        curr_timezone.zone
                     )
 
                 self._internal_end_date_of_backup = self._end_date_of_backup

